src/main.rs

```diff
@@ -50,10 +50,48 @@ ow=0:or=0;38;5;16;48;5;203:no=0:ex=1;38;5;203:cd=0;38;5;203;48;5;236:mi=0;38;5;1
 ";
 
+fn main() {
+    let result = run();
+    match result {
+        Ok(exit_code) => {
+            process::exit(exit_code.into());
+        }
+        Err(err) => {
+            eprintln!("[fd error]: {:#}", err);
+            process::exit(ExitCode::GeneralError.into());
+        }
+    }
+}
+
 fn run() -> Result<ExitCode> {
     let matches = app::build_app().get_matches_from(env::args_os());
 
-    // Set the current working directory of the process
+    set_working_dir(&matches)?;
+    let current_directory = Path::new(".");
+    ensure_current_directory_exists(&current_directory)?;
+    let search_paths = extract_search_paths(&matches, current_directory)?;
+
+    let pattern = extract_search_pattern(&matches)?;
+    check_pattern_is_not_path(&matches, pattern)?;
```

Comment:
Maybe also use "ensure_"  here, like above?

Reply:
Also, since we are concerned with readability here, let's go with: `ensure_that_search_pattern_is_not_a_path`?